Cip 88 v2 stake pools

Crypto2099 opened the pull request:
This CIP aims to extend and enhance CIP-0088 with a Version 2 that includes:

* New scope to register a stake pool
* New validation method to support CIP-0008/CIP-0030 (COSE Sign1) signing and witnesses
* Introduction of the Calidus Key

[Link to Rendered Version](https://github.com/Crypto2099/CIPs/tree/cip-88-v2-stake-pools/CIP-calidus-key)

Ryun1 commented on CIP-calidus-key/README.md:
```diff
@@ -0,0 +1,373 @@
+> There are currently no Stake Pool-specific CIPs to be used here but could be
+> added in the future.
+
+#### 7. Calidus Key
```
might be worth considering defining a derivation path for these keys

gitmachtl replied:
hey @Ryun1, yes this is something we can consider in an upgrade path. as those keys are hot/warm ones, they can be renewed fairly easy. 

we did not went with a fixed derivation path for now, because we would than also need support on the hw-wallet side for that too. and as financing/budget is currently in limbo mode, we did not want to generate extra costs or delays for now.

the suggestion currently is to use the standard payment path, and we have the tools on the cli that would also output mnemonics to reimport in a lightwallet for browser usage.

rphair reviewed (commented):
Great work @Crypto2099 @gitmachtl as agreed all around at Tuesday's CIP meeting.  Personally I look forward to feeding back any community review that comes in at the linked Forum location (and vice-versa, in case there's anything posted here of general SPO interest).  Ready of course to rename the directory & assign the new number in the header 🎉

rphair commented on CIP-calidus-key/README.md:
```diff
@@ -0,0 +1,373 @@
+  * [ ] PoolTool.io
+  * [ ] DripDropz
+* Registered Pools
+  * [ ] _x_ of _y_ SPOs have registered
```
@Crypto2099 @gitmachtl is there a quick Blockfrost query we could use to determine `x`?  (and maybe even `y`, for that matter?)

rphair replied:
Looks like we have one through Koios as per @gitmachtl https://github.com/cardano-foundation/CIPs/pull/999#issuecomment-2720539248 (apologies for my autistic query; feel free to correct with something properly semantic):
```
$ curl -s "https://api.koios.rest/api/beta/pool_calidus_keys" | jq | grep '"pool_status": "registered"' | wc -l
4
```
@Crypto2099 @Ryun1 @perturbing can we make an operating assumption about `y` or is there a quick & dirty way of determining that as well (which properly factors out multi-pool operators)?  cc @Balance-Analytics @ccgarant

gitmachtl replied:
you can do:

```
curl -s "https://api.koios.rest/api/beta/pool_calidus_keys?pool_status=eq.registered" | jq -r "length"
4
```

```
curl -s "https://preview.koios.rest/api/beta/pool_calidus_keys?pool_status=eq.registered" | jq -r "length"
6
```

but i am generally not sure if a metric x out of y pools must be used as an acceptance criteria.

rphair replied:
@Crypto2099 that's the way you had posted it (`x` out of `y`); are there any other measures of adoption we could consider along with this one?

rphair reviewed (commented):
As per my own requirements this looks ready for approval — and we can see some evidence from the Forum link of this being used in the wild — but I don't have the tooling / wallet / integration expertise to jump in & confirm the spec would be easily & enthusiastically followed.

So please review & post whether or not there's anything you would change about the document, plus any thoughts on implementations & of course any whistle blows about the specification itself (including the "key derivation" issue @Ryun1 mentioned in https://github.com/cardano-foundation/CIPs/pull/999#discussion_r1979861483): 

[a few from the _Acceptance_ checklist & please feel free to tag more] @KtorZ @MicroProofs @mmahut @rdlrt @Scitz0 @ashisherc @dmitrystas

Scitz0 commented:
A defined key derivation path would be the only thing as previously brought up to be able to use existing mnemonic phrases without interfering with other keys. I would suggest an extension to CIP-1853 with usecase `0`.

Other than that we have already made a first test implementation in Koios and I'm currently adding support in CNTools through @gitmachtl cardano-signer to be able to create keys and post calidus metadata.

gitmachtl commented:
> A defined key derivation path would be the only thing as previously brought up to be able to use existing mnemonic phrases without interfering with other keys. I would suggest an extension to CIP-1853 with usecase `0`.
> 
> Other than that we have already made a first test implementation in Koios and I'm currently adding support in CNTools through @gitmachtl cardano-signer to be able to create keys and post calidus metadata.

For me its important for all future cardano developments, that it should not exclude hardware wallets if possible. 
I was experimenting with the 1853'/1815' derivation path and i really don't wann use that because of the near relationship to the actual hw-cold-pool key on a hardware wallet. So that is 1853'/1815'/0'/0'. I would prefer to not use any kind of small variation to that tbh. For me this is to dangerous. 

Choosing a new derivation path would result in months of delay before we see proper hw-wallet implementation. The funding for ledger/trezor is currently in a limbo state, so i would prefer that we don't rely on an integration for that to merge this CIP.

Thats why we went with the standard payment path for now, every wallet can handle that. I also do the key generation with mnemonics as standard in the SPO scripts for that usecase, so SPOs can later simply copy&paste the mnemonics in a light wallet.

The good thing is, we can always upgrade to another path in the future and all that is required is that the SPO must renew the key.  Which should be a normal thing than for that key.

Scitz0 commented:
I dont feel HW support for Calidus key is necessary in this case as we are talking about a hot key that's easily rotatable.

I noticed that I had a typo in previous post. Updated usecase number to 1. 

Not hellbent on it though, I'm fine to skip it and keep CIP as is. Will most likely only use CLI keys anyway and this is what I implemented in CNTools.

gitmachtl commented:
> I noticed that I had a typo in previous post. Updated usecase number to 1.

Ah ok. So you mean 1853'/1815'/0'/1' or 1853'/1815'/0'/1 ?

Scitz0 commented:
After thinking some more about it I think to increase compatibility it's best to just stick to normal payment key path and CLI keys for now. Can revisit later if we feel there is a need for it.

gitmachtl commented:
We had a few updates this week, @Crypto2099 will update the docs ...

### Integration into the SPO scripting tools

* SPO Scripts do have Calidus Pool-Key support now, so this can be checkmarked in the doc
* CN-Tools has Calidus Pool-Key support now in the alpha branch

### Calidus-ID

After talks with @Scitz0 and other lightwallet vendors, it was clear that we need kind of an `address` for the `dataSign CIP-0030` process. So the `Calidus-ID` was created pretty similar to the DRep-ID/CC-Cold-ID/CC-Hot-ID we know from CIP-129.
The `Calidus-ID` is the bech representation of the Calidus-PublicKey hash with an extra prebyte `0xa1` to identify it as Calidus **Pool** Key, because we might see updates in the future.

```ini
calidusPubKeyHash = blake2b( calidusPubKey, 224 ) //blake2b hash with 224 bits
calidusIdHex = `a1${calidusPubKeyHash}` //add the hex-prebyte a1 to indentify as pool-key
calidusIdBech = bechEncode( calidusIdHex, `calidus` ) //convert into bech with the prefix 'calidus'
``` 

The choosen bech prefix is `calidus`, so a valid calidus-id looks like
```json
{
    "pool_id_bech32": "pool1rdaxrw3722f0x3nx4uam9u9c6dh9qqd2g83r2uyllf53qmmj5uu",
    "pool_status": "registered",
    "calidus_nonce": 150084961,
    "calidus_pub_key": "2819290ac05b4a49331d1781ce1dfc04cddfa8875abee8303ea5311878f12daf",
    "calidus_id_bech32": "calidus15xcc3mphdwu8amrx9m6v9v4552pugr2gcsj4m8arjmx3zwc8yf8ng",
    "tx_hash": "1c51a6543c1fa8d70620fdba3469d8e632626aa85e9f09896825a59fcba188bf",
    "epoch_no": 635,
    "block_height": 2725004,
    "block_time": 1741683427
  }
```

### Skey and Vkey files on the CLI

A decision was made to use the generic payment path `1852H/1815H/0H/0/0` for now. To identify a calidus key file, the description field was changed to be `Calidus Pool Signing Key` & `Calidus Pool Verification Key`. The `type` field was left untouched to be compatible with other cli tools.

A typical key pair looks like
```json
calidus.skey
{
  "type": "PaymentExtendedSigningKeyShelley_ed25519_bip32",
  "description": "Calidus Pool Signing Key",
  "cborHex": "588000b39b6802b11c2c5043dbdccd6d308d9d0e44b619b1c64bd135bade997661538840b086598855186dd6c7abc7cffadc71f0409d9a7c346f61e139569a85829c2819290ac05b4a49331d1781ce1dfc04cddfa8875abee8303ea5311878f12daf413e4f538ba125596407fff31a087e994956e199c7de2e956f511c1917e3f944"
}

calidus.vkey
{
  "type": "PaymentVerificationKeyShelley_ed25519",
  "description": "Calidus Pool Verification Key",
  "cborHex": "58202819290ac05b4a49331d1781ce1dfc04cddfa8875abee8303ea5311878f12daf"
}
```

### Cardano-Signer

* Cardano-Signer v1.24.0 now automatically generated those kind of keyfiles if launched with the new shortcut path for calidus `--path calidus` like

```console
cardano-signer sign --cip88 --path calidus --out-skey calidus.skey --out-vkey calidus.vkey --json-extended | jq .
``` 

### Updated FlowChart

The new version now also shows the generation of the `Calidus-ID`
![image](https://github.com/user-attachments/assets/3dd78534-d9e4-4033-90a3-fce682378aa2)

### Updates from the light wallets

* eternl is working on an update to enable `dataSign` for `calidus1xxx` addresses (@Scitz0)
* typhon wallet is also working on the same update, should be available within next week (@ashisherc)

### Updates on koios

* We have now full support for the calidus-pool-key query on `Sancho-Net`. The query can be accessed like
```console
curl -s "https://sancho.koios.rest/api/v1/pool_calidus_keys" | jq .
```

gitmachtl commented:
### Updates on koios
* We have now a Mainnet-Beta endpoint for the calidus-pool-key query
```console
curl -s "https://api.koios.rest/api/beta/pool_calidus_keys" | jq .
```

Crypto2099 commented:
I have updated the documentation to include the bech32 prefixing/encoding information and add some updated implementers. I've also updated to @gitmachtl newer flowchart that was shared here in the comments.

Ryun1 commented on CIP-0151/README.md:
```diff
@@ -0,0 +1,434 @@
+---
+CIP: 0151
```
```suggestion
CIP: 151
```

rphair reviewed (approved):
@Crypto2099 @gitmachtl from what I've seen in the community this is certainly suitable as `Proposed` and looks well on its way to being `Active` (except not as defined conservatively in the _Path to Active_ yet).

@Crypto2099 if you think this is not ready for merge then please take it down from `Last Check` but otherwise I think we should go ahead and merge this at the next CIP meeting (@gitmachtl as always you are welcome to come): https://hackmd.io/@cip-editors/110

rphair commented:
NOTE BEFORE MERGE (though I'll leave my ✅ up since I'm still happy with the content itself) that we need to confirm the title before merging.  If @gitmachtl's enquiry to @Crypto2099 in https://github.com/cardano-foundation/CIPs/pull/1007#issuecomment-2796525013 isn't settled before the next CIP meeting then we should settle the title issue then.

Crypto2099 commented:
A couple of relatively minor updates this morning as we prepare to have this CIP merged into "proposed" status. Renamed to match other in-flight CIPs at @gitmachtl suggestion.

rphair reviewed (approved):
still looks good: updating PR title to match the new document title (same except for the plural): `On-Chain Registration - Stake Pools`

Ryun1 reviewed (approved):
really well written and motivated proposal
get work @Crypto2099 @gitmachtl
